projects/plugins/jetpack/extensions/plugins/publicize/store/reducer.js

```diff
@@ -13,15 +13,57 @@ export const DEFAULT_STATE = {
  */
 export default function ( state = DEFAULT_STATE, action ) {
 	switch ( action.type ) {
-		case 'SET_CONNECTION_TEST_RESULTS':
+		case 'SET_CONNECTION_TEST_RESULTS': {
+			// Combine current connections with new connections.
+			const { connections: prevConnections } = state;
+			const prevConnectionIds = prevConnections.map( connection => connection.id );
+
+			const { results: freshConnections } = action;
+
+			const connections = [];
+			for ( const freshConnection of freshConnections ) {
+				let connection;
+				if ( prevConnectionIds.includes( freshConnection.id ) ) {
+					/*
+					 * The connection is already defined.
+					 * Do not overwrite the existing connection.
+					 */
+					connection = prevConnections.filter(
```

Comment:
This probably doesn't matter as there's only ever a few connections, but I think something that doesn't involve a full scan of the array would be better here, like `findIndex`. We could also remove the need for `prevConnectionIds`